### PR TITLE
Add endpoint for surveys and add tests Co-authored-by: Daniel Nottingham

### LINKED ISSUE #156: if user is not admin he shouldn't get all the surveys
update endpoint for surveys by user_Id

### PR DESCRIPTION
**Pull request by garciajordy**
# Surveys Scope

**Survey Model:**
- Add a scope to only show surveys created by the user when the user is not an admin.

**SurveysController:**
- Add our brand new scope in the index method.

**Rspec:**
- Add test when the user is an admin.
- Add test when the user is a teacher.
- Add test when the user is a moderator.

_Thanks for taking the time to review our PR._ 😄 

### _Co-authored-by: Daniel Nottingham <danielnottingham@users.noreply.github.com>_
fix #156

### CODE REVIEW

File: backend/Gemfile [hunk 1]
```diff
@@ -39,7 +39,7 @@ group :development, :test do
   gem 'bullet'
   gem 'faker', :git => 'https://github.com/faker-ruby/faker.git', :branch => 'master'
 end
-
+#
```
Comment (edimossilva) [inline backend/Gemfile:42]: remove this?

File: backend/spec/models/survey_spec.rb [hunk 2]
```diff
@@ -5,4 +5,16 @@
     it { is_expected.to belong_to(:user).required }
     it { is_expected.to have_many(:questions) }
   end
+
+  describe '.by_user' do
```
Comment (edimossilva) [inline backend/spec/models/survey_spec.rb:9]: Also, need specs for the controller. They will be similar to these one, but simpler

**Review by edimossilva — CHANGES_REQUESTED** [review]
@garciajordy and @danielnottingham

File: backend/Gemfile (hunk 1 above)
Comment (garciajordy) [reply backend/Gemfile:42]: Oh, that was probably when writing in live share, will fix this!

File: backend/spec/models/survey_spec.rb (hunk 2 above)
Comment (garciajordy) [reply backend/spec/models/survey_spec.rb:9]: Ok! We will add those as well!

**Comment by garciajordy** [comment]
# Changes!

**Gemfile:**
- Remove unnecessary "#".

**Rspec:**
- Add integration tests in surveys_crud_spec.rb file.

_Thank you for taking the time to review our changes!_ 😄

**Review by edimossilva — CHANGES_REQUESTED** [review]
I believe you should not add system specs. Since it is not related to the current code changes.

The code that you guys updated is not used by system specs. The administrate gem manages everything related to that for us.

The code that you guys updated is only used by API and currently can only be tested by `requests specs` or `controller specs` (for this project we choose to use `requests specs`).

You should work on this file `https://github.com/espoo-dev/espoo-dev/blob/main/backend/spec/requests/surveys_request_spec.rb`

**Comment by edimossilva** [comment]
@garciajordy , I will close this PR because @danielnottingham  did the fixes that I was needing for my other task

**Pull request closed without merging**